lib/azure/armrest/storage/managed_storage_helper.rb

```diff
@@ -58,28 +58,46 @@ def get_blob_raw(disk_name, resource_group = configuration.resource_group, optio
     }
 
     # This call will give us an operations URL in the headers.
-    initial_url = build_url(resource_group, disk_name, 'BeginGetAccess')
-    response    = rest_post(initial_url, post_options.to_json)
+    begin_get_access_url      = build_url(resource_group, disk_name, 'BeginGetAccess')
+    begin_get_access_response = rest_post(begin_get_access_url, post_options.to_json)
+
     begin
-      headers = Azure::Armrest::ResponseHeaders.new(response.headers)
+      headers = Azure::Armrest::ResponseHeaders.new(begin_get_access_response.headers)
 
       # Using the URL returned from the above call, make another call that
       # will return the URL + SAS token.
       op_url = headers.try(:azure_asyncoperation) || headers.location
 
       unless op_url
         msg = "Unable to find an operations URL for #{disk_name}/#{resource_group}"
-        raise Azure::Armrest::NotFoundException.new(response.code, msg, response.body)
+        log('debug', "#{msg}: #{begin_get_access_response.headers}")
+        raise Azure::Armrest::NotFoundException.new(begin_get_access_response.code, msg, begin_get_access_response.body)
       end
 
       # Dig the URL + SAS token URL out of the response
       response = rest_get(op_url)
       body     = Azure::Armrest::ResponseBody.new(response.body)
       sas_url  = body.try(:properties).try(:output).try(:access_sas)
 
-      unless sas_url
-        msg = "Unable to find an SAS URL for #{disk_name}/#{resource_group}"
-        raise Azure::Armrest::NotFoundException.new(response.code, msg, response.body)
+      max_sas_url_retries = 20
+      sas_url_retries     = 0
+
```

Comment:
this whole loop looks overly complicated and hard to read.  Can we extract it to a new method called `wait_for_async_operation` where all the trying and retrying is done?

Reply:
@chessbyte let me discuss with @djberg96 regarding his previous comment and see what we can do to either simplify or extract as you suggest.